### DISCUSSION TITLE
Setting up Shiny services for MAUI

### DISCUSSION THREAD

**Discussion by tongvietdung** [shinyorg/shiny#1129] [Q&A] [answered]
I'm following this changelog http://shinylib.net/release-notes/mobile/vnext.html to setup for MAUI. 
Installed packages: Shiny, Shiny.Core, Shiny.Hosting.Maui, Shiny.SpeechRecognition. 
I've came across these error when setting up:
Error at line ```.UseShiny()```
>'MauiAppBuilder' does not contain a definition for 'UseShiny' and no accessible extension method 'UseShiny' accepting a first argument of type 'MauiAppBuilder' could be found (are you missing a using directive or an assembly reference?)

Error at line ```builder.Services.AddSpeechRecognition();```
>'IServiceCollection' does not contain a definition for 'AddSpeechRecognition' and no accessible extension method 'AddSpeechRecognition' accepting a first argument of type 'IServiceCollection' could be found (are you missing a using directive or an assembly reference?)

My ```MauiProgram.cs```
```
using Shiny;

namespace SpeechRecognitionTest;

public static class MauiProgram
{
	public static MauiApp CreateMauiApp()
	{
		var builder = MauiApp.CreateBuilder();
		builder
			.UseMauiApp<App>()
			.UseShiny()
			.ConfigureFonts(fonts =>
			{
				fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
				fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
			});

		builder.Services.AddSpeechRecognition();
        return builder.Build();
	}
}
```

Please help me resolve this problem. Thank you.

**Comment by aritchie** [comment, marked as answer]
My suggestion is not to use the alphas if you're having trouble.  You can look the samples in the master branch or checkout https://github.com/aritchie/BeatTheBank

Since the "Shiny" nuget package isn't currently used in v3, my guess is you're not using the proper version for maui

**Reply by tongvietdung** [reply]
Thank you for your reply. 
Btw, Is it possible in the future that Speech Recognition Service can transcribe audio file into text?

**Reply by aritchie** [reply]
No.  It uses the mic

**Reply by tongvietdung** [reply]
Thank you. I've taken a look at your BeatTheBank and made it work in my personal project. I really had fun with it xD
However it sometimes does not get the sentence or missing some 1-2 last words. What model do you use for this speech recognition service?

**Reply by aritchie** [reply]
I'm going to suggest you take a look at the code to get a better understanding of everything
